### ISSUE TITLE
Proxy Protocol for UDP

### ISSUE THREAD

**Issue by cleverer** [traefik/traefik#8613]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you expect to see?

I'm trying to set up a PowerDNS server running completely behind traefik proxy. From version 4.6.0 upwards, PowerDNS supports Proxy Protocol headers, which would be quite essential in the DNS UDP world (to support AXFR requests among other things). PowerDNS currently is unable to see the remote IP on UDP packets, as it only sees traefiks IP.

Similarly to #5371, I'd like to be able to set proxyProtocol on udp services.

**Comment by 3nprob** [comment]
@traefiker don't auto-close

**Comment by jjsaunier** [comment]
Any news on this? That prevents ip filtering on TURNserver (webrtc)

**Comment by emilevauge** [comment]
Any contribution on this topic is welcome :)

**Comment by jjsaunier** [comment]
ok, but that's not an easy topic, since there is no "standard" yet, some choices must be made and that must be cleared by the core team before. 

Currently Cloudflare have something called "Simple Proxy Protocol" to address this topic https://developers.cloudflare.com/spectrum/reference/simple-proxy-protocol-header/#payload and there is also an issue on the current proxyproto package used for proxy protocol v1/v2 https://github.com/pires/go-proxyproto/issues/88 to address it

I don't know if with the raise of http3/quic new protocol proposal will emerge

**Comment by moschny** [comment]
> since there is no "standard" yet

Am I missing something, or wouldn't that also be the ha proxy protocol?
It seems that e.g. knotd explicitly supports it for UDP, see https://www.knot-dns.cz/docs/3.3/singlehtml/index.html#proxy-allowlist

**Comment by 4310V343k** [comment]
ProxyProtocolV2 supports UDP

**Comment by ghost** [comment]
Is there a way we can adapt this?

This has a functioning reverse udp proxy

https://github.com/mholt/caddy-l4/blob/master/modules/l4proxy/proxy.go

**Comment by errordlien** [comment]
I've created https://github.com/traefik/traefik/pull/12054 based off the caddy-l4 implementation.

Hopefully we can get some traction or additional eyes on it and get it implemented soon.

**Comment by t3cneo** [comment]
Hi 

#12224 looks like its adding UDP Poxy Protocol to traefik as a destination (behind another reverse proxy that sends Proxy Protocol UDP packets)

#8613 to my understanding is about adding UDP Proxy Protocol to traefik as a source (when traefik is in front of UDP services supporting UDP Proxy Protocol such as DNS or TURN)

As OP @cleverer I'm also trying to setup a DNS server behind traefik and was brought here when I realised traefik can't handle UDP Proxy Protocol to services

Being able to get real IP for UDP services (supporting Proxy Protocol) behind traefik would be an absolute banger

**Comment by polarathene** [comment]
I think Caddy L4 plugin might have Proxy Protocol UDP support, you could give that a look?

It's a bit verbose (and no use of Proxy Protocol), but might be a useful reference for you in setting up Caddy L4 for DNS-over-TLS proxy: https://github.com/mholt/caddy-l4/issues/276#issuecomment-2817496982

That issue has [another comment with just Caddy L4 config](https://github.com/mholt/caddy-l4/issues/276#issuecomment-3538440235) rather than setting everything up, so it's not as bad as my verbose example looks 😅

**Comment by t3cneo** [comment]
Yep I know both caddy and nginx are able to send UDP Proxy Protocol v2 as a reverse proxy but traefik won my heart years ago and is routing 50+ public facing production-use services, trust in myself is too small to try and switch overnight 😅

**Comment by sdelicata** [comment]
@t3cneo You're correct #12224 handles reception (upstream → Traefik). 
I plan to add transmission (Traefik → backends) when I have time to complete the use case.

**Comment by t3cneo** [comment]
@sdelicata great news, thanks for your feedback and hard work. Can't wait to finally be able to put everything behind traefik :innocent:

**Comment by DanielAdolfsson** [comment]
@sdelicata 
Feels like the PR have ended up in a bit of a limbo.
As I'm in need of this, is there anything I can do to help?

**Comment by felinusfish** [comment]
Bump, this feels strange to have been stuck in limbo for so long. I've been in desperate need for this for game servers.
